src/test/java/core/basesyntax/RegistrationServiceTest.java

```diff
@@ -0,0 +1,165 @@
+        assertEquals(actual, john);
+        assertEquals(nextUserToAdd, maria);
+    }
+
+    @Test
+    void registerNotAdult_NotOk() {
+        john.setAge(TEEN_AGE);
+        john.setLogin(validLogin);
+        john.setPassword(validPassword);
+        assertThrows(InvalidDataException.class, () -> registrationService.register(john));
+    }
+
+    @Test
+    void inadmissibleAge_NotOK() {
+        john.setAge(null);
+        john.setLogin(validLogin);
+        john.setPassword(validPassword);
+        george.setAge(ZERO_AGE);
+        george.setLogin(anotherValidLogin);
+        george.setPassword(anotherValidPassword);
+        jane.setAge(NEGATIVE_AGE);
+        jane.setLogin(thirdValidLogin);
+        jane.setPassword(validPassword);
+        assertThrows(InvalidDataException.class, () -> registrationService.register(john));
+        assertThrows(InvalidDataException.class, () -> registrationService.register(george));
+        assertThrows(InvalidDataException.class, () -> registrationService.register(jane));
+    }
+
+    @Test
+    void shortLogin_NotOk() {
+        john.setAge(ADULT_AGE);
+        john.setLogin(invalidLogin);
+        john.setPassword(validPassword);
+        assertThrows(InvalidDataException.class, () -> registrationService.register(john));
+    }
+
+    @Test
+    void shortPassword_NotOk() {
+        john.setAge(ANOTHER_ADULT_AGE);
+        john.setLogin(validLogin);
+        john.setPassword(invalidPassword);
+        assertThrows(InvalidDataException.class, () -> registrationService.register(john));
+    }
+
+    @Test
+    void emptyLogin_NotOk() {
+        john.setAge(ADULT_AGE);
+        john.setLogin("");
+        john.setPassword(validPassword);
+        assertThrows(InvalidDataException.class, () -> registrationService.register(john));
+    }
+
+    @Test
+    void emptyPassword_NotOK() {
+        john.setAge(ADULT_AGE);
+        john.setLogin(validLogin);
+        john.setPassword("");
+        assertThrows(InvalidDataException.class, () -> registrationService.register(john));
+    }
+
+    @Test
+    void nullLogin_NotOk() {
+        john.setAge(ANOTHER_ADULT_AGE);
+        john.setLogin(null);
+        john.setPassword(validPassword);
+        assertThrows(InvalidDataException.class, () -> registrationService.register(john));
+    }
+
+    @Test
+    void nullPassword_NotOk() {
+        john.setAge(ADULT_AGE);
+        john.setLogin(validLogin);
+        john.setPassword(null);
+        assertThrows(InvalidDataException.class, () -> registrationService.register(john));
+    }
+
+    @Test
+    void registerTheSameUser_NotOk() {
+        john.setAge(ANOTHER_ADULT_AGE);
+        john.setLogin(validLogin);
+        john.setPassword(validPassword);
+        Storage.people.add(john);
+        assertThrows(InvalidDataException.class, () -> registrationService.register(john));
+    }
+
+    @Test
+    void registerAnotherUserWithSameLogin_NotOk() {
```

Comment:
This test method 'registerAnotherUserWithSameLogin_NotOk' is not correct because it relies on the 'register' method to add a user to the storage for testing. Instead, use a low-level method like 'Storage.people.add()' to add users for testing. Also, rename the test to follow the naming convention, such as 'register_userWithExistingLogin_NotOk'.